proposals/2022-09-14-kubernetes-versioning.md

```diff
@@ -0,0 +1,208 @@
+# Wasm Shim Versioning in Kubernetes
+Wasm workloads in Kubernetes will likely need to lock into a specific version of a Wasm runtime. This proposal is intended to address this problem.
+
+
+## <a name='TableofContents'></a>Table of Contents
+
+<!-- vscode-markdown-toc -->
+* [Table of Contents](#TableofContents)
+* [Background](#Background)
+* [Problem Statement](#ProblemStatement)
+	* [Goals](#Goals)
+	* [Non Goals](#NonGoals)
+* [Proposal](#Proposal)
+	* [Versioning of both shim binary and RuntimeClass name](#VersioningofbothshimbinaryandRuntimeClassname)
+	* [Introducing a new version of a shim](#Introducinganewversionofashim)
+* [Alternatives](#Alternatives)
+	* [Alternative #1](#Alternative1)
+		* [Pros](#Pros)
+		* [Cons](#Cons)
+		* [Conclusion](#Conclusion)
+* [Additional Details](#AdditionalDetails)
+	* [Test Plan](#TestPlan)
+* [Implementation History](#ImplementationHistory)
+
+<!-- vscode-markdown-toc-config
+	numbering=false
+	autoSave=false
+	/vscode-markdown-toc-config -->
+<!-- /vscode-markdown-toc -->
+
+## <a name='Background'></a>Background
+
+It is likely behaviors in a shim may change over time as new features are introduced or as existing defects are resolved. Pod workloads that target a version N of a shim may not be able to execute on version N+1. This proposal describes a solution to ensure Pod workloads targeting a specific version of a shim will always be scheduled on a node that contains a safe version of a shim for the workload.
```

Comment:
this `Pod workloads that target a version N of a shim may not be able to execute on version N+1` should be `Pod workloads that target a version N of a shim may not be able to execute on version N-1` we have to at least guarantee that forward compatibility.

Reply:
I was referring to major version increments. I need to be more precise here.